draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -831,16 +834,82 @@ Instantiating one of these protocols with an elliptic curve group motivates
 the term "hashing to the curve", i.e., encoding bitstrings to points on an
 elliptic curve.
 
-The hash2curve(alpha) function can be easily constructed by
-using the output of a cryptographically secure hash function H as the input of
-an encoding function.
-On the one hand, hash2curve is difficult to invert since it is computationally
-intractable to produce an input alpha that maps to hash2curve(alpha) due to H
-is pre-image resistant.
+One can easily construct a hash to curve function h(x) by using the output of a
+cryptographically secure hash function H as the input of an encoding function.
+On the one hand, h(x) is difficult to invert since it is computationally
+intractable to produce an input x that maps to h(x) due to H is pre-image
+resistant.
 On the other hand, the uniformity property is not met as the output of an
 encoding is distinguishable from a random distribution. Hence, using
-hash2curve(alpha) is not sufficient to get uniformity, however it can be used
-as a building block for obtaining a random oracle as is described in {{rom}}.
+h(x) is not sufficient to get uniformity and cannot be used as a random oracle.
+
+Brier et al. {{BCIMRT10}} describe two generic constructions that give a
+hash function approximating a random oracle. Farashahi et al. {{FFSTV13}}
+and Tibouchi and Kim {{TK17}} refine this analysis. In particular, Farashahi
+et al. show that summing two independent evaluations of any of the
+deterministic endings of {{encodings}} suffices to approximate a random
+oracle to an elliptic curve. In other words:
+
+~~~
+   h(x) = f(H0(x)) + f(H1(x))
+~~~
+
+where f: F -> E(F) is a deterministic encoding, H0 and H1 are hash
+functions (modeled as random oracles) mapping strings to elements of F, and the
+addition operation corresponds to point addition on the target elliptic curve.
+
+
+# Roadmap on Hash to Curve. {#roadmap}
+
+The following functions provide a general framework for mapping bit
+strings into points of a prime group G.
+
+ - Injective encodings. A bit string is mapped to a point in G, however,
+   the distribution of this function is not uniform.
+
+   ~~~
+   h2cInjective(alpha)
+
+   Input: alpha, an arbitrary-length bit string.
+   Output: P, a point in the group G.
+
+   Steps:
+   1. u = hash2base(alpha, I2OSP(0, 1)) // {0,1}* -> F   
+   2. Q = map2curve(u)                  //   F    -> E(F)
+   3. P = clearCofactor(Q)              //  E(F)  -> G   
```

Comment:
Naming related: this should probably be `clear_cofactor` to be consistent with other symbol names.

Reply:
Agreed!